Lesson15/Ayse/script.js

```diff
@@ -0,0 +1,504 @@
+  console.log(censoredSentence);
+}
+
+censorWord('I love apples and apples are great.', 'apples');
+
+/*
+15. Check First Character (if-else)
+   - Define a function `startsWithA(str)` that checks if the string starts with 'A'
+     (use .charAt(0) or [0]).
+   - Log "Starts with A" or "Does not start with A".
+*/
+console.log("Example-15");
+
+function startsWithA(str) {
+  if (str.charAt(0) == 'A') {
+    console.log('Starts with A');
+  } else {
+    console.log('Does not start with A');
+  }
+}
+startsWithA('Apple');
+startsWithA('Banana');
+
+/*
+16. Slice Last N Characters
+   - Define a function `sliceLastN(text, n)` that uses .slice(-n) to extract
+     the last `n` characters of `text`.
+   - Log the result.
+*/
+console.log("Example-16");
+
+function sliceLastN(text, n) {
+  const result = text.slice(-n);
+  console.log(result);
+}
+
+sliceLastN('Hello', 3);
+sliceLastN("JavaScript", 4);
+sliceLastN("Hello, World!", 1);
+
+/*
+17. Switch: Grade Checker
+   - Define a function `gradeChecker(score)` that uses a switch (or if-else chain):
+     90+ -> "A"
+     80-89 -> "B"
+     70-79 -> "C"
+     60-69 -> "D"
+     below 60 -> "F"
+   - Log the grade.
+*/
+console.log("Example-17");
+
+function gradeChecker(score) {
+  switch (true) {
+    case (score >= 90):
+      console.log('A');
+      break;
+    case (score >= 80 && score <= 89):
+      console.log('B');
+      break;
+    case (score >= 70 && score <= 79):
+      console.log('C');
+      break;
+    case (score >= 60 && score <= 69):
+      console.log('D');
+      break;
+    case (score < 60):
+      console.log('F');
+      break;
+    default:
+      console.log('Invalid score');
+  }
+}
+gradeChecker(100);
+gradeChecker(85);
+
+/*
+18. Character Replacement
+   - Define a function `replaceCharacter(str, oldChar, newChar)` that uses .replaceAll()
+     (or a loop) to swap all occurrences of oldChar with newChar.
+   - Log the result.
+*/
+console.log("Example-18");
+
+function replaceCharacter(str, oldChar, newChar) {
+  const replacedResult = str.replaceAll(oldChar, newChar);
+  console.log(replacedResult);
+}
+
+replaceCharacter('Hello', 'l', 'y');
+
+/*
+19. Title Case a Sentence
+   - Define a function `titleCase(sentence)` that:
+     - Splits the sentence by spaces
+     - Uppercases the first letter of each word
+     - Joins them back
+   - Log the transformed string.
+*/
+console.log("Example-19");
+
+function titleCase(sentence) {
+  const titleCased = sentence.split(' ').map(word => word.charAt(0).toUpperCase() + word.slice(1).toLowerCase()).join(' ');
+  console.log(titleCased);
+}
+titleCase('hello world!');
+
+
+/*
+20. Switch: Traffic Light
+   - Define a function `trafficLight(color)` that uses a switch statement:
+     - "red" -> log: "Stop"
+     - "yellow" -> log: "Caution"
+     - "green" -> log: "Go"
+     - anything else -> "Invalid color"
+*/
+console.log("Example-20");
+
+function trafficLight(color) {
+  switch (color) {
```

Comment:
This switch case is a good example of situation where switch case code is cleaner than if-else statement.